nautobot/core/api/fields.py

```diff
@@ -228,8 +228,8 @@ def to_representation(self, value):
             model = self._related_model
 
         if model is None:
-            return {"id": value.pk, "object_type": "unknown.unknown", "url": url}
-        return {"id": value.pk, "object_type": model._meta.label_lower, "url": url}
+            return {"id": str(value.pk), "object_type": "unknown.unknown", "url": url}
+        return {"id": str(value.pk), "object_type": model._meta.label_lower, "url": url}
```

Comment:
In theory, `to_representation` should only yield JSON serialisable data, but `UUID` is not JSON serialisable, therefore `publish_event` cannot accept a non-JSON serialisable object.